### DISCUSSION TITLE
How to set the contact thermal resistance and radiation heat transfer on the surface of a continuous block?

### DISCUSSION THREAD

**Discussion by LeonDong1998** [idaholab/moose#28592] [Q&A Modules: General]
Hello everyone!

I want to know how continuous surface heat transfer can achieve a similar effect to GapHeatTransfer, the  emissivity of two contact surfaces can be set. And although there is no thermal conductivity of the gap material, the contact thermal resistance can be set.

Thank you in advance!

**Comment by GiudGiud** [comment]
Hello

If you dont have a gap in the mesh but instead have a continuous surface, then you can use an interface kernel.
This one should do it
https://mooseframework.inl.gov/source/interfacekernels/SideSetHeatTransferKernel.html

Guillaume

**Reply by LeonDong1998** [reply]
Thanks a lot! 

And I tried to build a simulation using SideSetHeatTransfer Kernel, but encountered the following error:
"Variable and neighbor variable are the same, but they are not elemental variables."

Afterwards, I changed the family from LAGRANGE to MONOMIN, and the following error occurred, which may be related to my calculation coupling openmc:
"max_T: cannot couple elemental variables into nodal objects"

Thank you in advance!

**Reply by GiudGiud** [reply]
The first object is refusing to do an interface treatment for a continuous nodal variable. This makes sense you might have to actually create the gap using a BreakMesh...Generator then use a proper gap heat transfer treatment

max_T is a postprocessor. This is a different object giving you the error. 
You probably have a NodalExtremeValue postprocessor which wont accept an elemental variable
